src/btree/bt_compact.c

```diff
@@ -340,8 +342,9 @@ __wt_compact(WT_SESSION_IMPL *session)
          * Periodically check if we've timed out or eviction is stuck. Quit if eviction is stuck,
          * we're making the problem worse.
          */
-        if (++i > 100) {
-            bm->compact_progress(bm, session, &msg_count);
+        if (count == 0 || ++i > 100) {
+            if (count > 0)
+                bm->compact_progress(bm, session, &msg_count);
```

Comment:
It looks like the only use of `count` is to identify the first pass through this loop. So you could just use a boolean.